hbase-server/src/main/java/org/apache/hadoop/hbase/quotas/QuotaUtil.java

```diff
@@ -266,6 +266,13 @@ private static void deleteQuotas(final Connection connection, final byte[] rowKe
     if (qualifier != null) {
       delete.addColumns(QUOTA_FAMILY_INFO, qualifier);
     }
+    if (isNamespaceRowKey(rowKey)) {
+      String ns = getNamespaceFromRowKey(rowKey);
+      Quotas namespaceQuota = getNamespaceQuota(connection,ns);
+      if (namespaceQuota != null && namespaceQuota.hasSpace()) {
+        deleteTableUsageSnapshotsForNamespace(connection, ns);
```

Comment:
Please add a comment here as to why we need to delete the table usage rows (to prevent the next person from having to come back and do the same investigation we did :)

Reply:
Will do, thank you.